File: algos/graph_py/cmd/daily_gen.py
#!/usr/bin/env python3
import os
from datetime import datetime


def main():
    # Get current date in YYYYMMDD format
    today = datetime.now()
    date_str = today.strftime("%Y%m%d")

    # Define template configurations
    templates = {
        "traverse": {"template_file": "traverse_template.tmpl", "output_prefix": "traverse"},
        "dag": {"template_file": "dag_template.tmpl", "output_prefix": "dag"},
        "dijkstra": {"template_file": "dijkstra_template.tmpl", "output_prefix": "dijkstra"},
    }

    generated_count = 0
    generated_files = []

    # Generate all templates
    for template_type, config in templates.items():
        # Create output filename
        output_file = f"{config['output_prefix']}_{date_str}.py"

        # Check if file already exists in parent directory
        output_path = os.path.join("..", output_file)
        if os.path.exists(output_path):
            print(f"File {output_file} already exists for today")
            continue

        # Read template file
        template_path = os.path.join("..", config['template_file'])
        try:
            with open(template_path, 'r') as template_file:
                template_content = template_file.read()
        except FileNotFoundError:
            print(f"⚠️  Warning: Could not open template file {config['template_file']}")
            continue

        # Create output file in parent directory
        try:
            with open(output_path, 'w') as output_file_handle:
                # Add header comment
                header = f"# Generated on {today.strftime('%Y-%m-%d %H:%M:%S')}\n"
                header += f"# Daily practice file: {output_file} ({template_type} template)\n\n"
                output_file_handle.write(header)

                # Write template content to output file
                output_file_handle.write(template_content)

            generated_files.append(output_file)
            generated_count += 1
        except IOError as e:
            print(f"⚠️  Warning: Could not create output file {output_file}: {e}")
            continue

    # Print summary
    if generated_count == 0:
        print("No new files generated - all templates already exist for today")
    else:
        print(f"✅ Successfully generated {generated_count} file(s) for today's practice:")
        for file in generated_files:
            print(f"   - {file}")


if __name__ == "__main__":
    main()

File: algos/archived/graph/cmd/daily_gen.go
package main

import (
	"fmt"
	"io"
	"os"
	"time"
)

func main() {
	// Get current date in YYYYMMDD format
	today := time.Now()
	dateStr := today.Format("20060102")

	// Define template configurations
	templates := map[string]struct {
		templateFile string
		outputPrefix string
	}{
		"traverse": {"traverse_template.tmpl", "traverse"},
		"dag":      {"dag_template.tmpl", "dag"},
		"dijkstra": {"dijkstra_template.tmpl", "dijkstra"},
	}

	generatedCount := 0
	var generatedFiles []string

	// Generate all templates
	for templateType, config := range templates {
		// Create output filename
		outputFile := fmt.Sprintf("%s_%s.go", config.outputPrefix, dateStr)

		// Check if file already exists in current directory
		outputPath := outputFile
		if _, err := os.Stat(outputPath); err == nil {
			fmt.Printf("File %s already exists for today\n", outputFile)
			continue
		}

		// Read template file
		templateFile, err := os.Open(config.templateFile)
		if err != nil {
			fmt.Printf("⚠️  Warning: Could not open template file %s: %v\n", config.templateFile, err)
			continue
		}

		// Create output file
		outputFileHandle, err := os.Create(outputPath)
		if err != nil {
			templateFile.Close()
			fmt.Printf("⚠️  Warning: Could not create output file %s: %v\n", outputPath, err)
			continue
		}

		// Add header comment
		header := fmt.Sprintf("// Generated on %s\n// Daily practice file: %s (%s template)\n\n",
			today.Format("2006-01-02 15:04:05"), outputFile, templateType)
		outputFileHandle.WriteString(header)

		// Copy template content to output file
		_, err = io.Copy(outputFileHandle, templateFile)
		templateFile.Close()
		outputFileHandle.Close()

		if err != nil {
			fmt.Printf("⚠️  Warning: Error copying template to output file %s: %v\n", outputFile, err)
			continue
		}

		generatedFiles = append(generatedFiles, outputFile)
		generatedCount++
	}

	// Print summary
	if generatedCount == 0 {
		fmt.Println("No new files generated - all templates already exist for today")
	} else {
		fmt.Printf("✅ Successfully generated %d file(s) for today's practice:\n", generatedCount)
		for _, file := range generatedFiles {
			fmt.Printf("   - %s\n", file)
		}
	}
}
